tests/test_renault_vehicle.py

```diff
@@ -98,6 +100,25 @@ async def test_get_hvac_status(vehicle: RenaultVehicle) -> None:
         assert await vehicle.get_hvac_status()
 
 
+@pytest.mark.asyncio
+async def test_get_hvac_settings(vehicle: RenaultVehicle) -> None:
+    """Test get_hvac_settings."""
+    with aioresponses() as mocked_responses:
+        mocked_responses.get(
+            f"{TEST_KAMEREON_VEHICLE_URL1}/hvac-settings?{QUERY_STRING}",
+            status=200,
+            body=get_file_content(f"{FIXTURE_PATH}/vehicle_data/hvac-settings.json"),
+        )
+        data = await vehicle.get_hvac_settings()
+
+        assert data.mode == "scheduled"
+        for i in range(1, 6):
+            assert data.schedules[i - 1].activated is False
```

Comment:
For Optional types, you need to ensure it is not None before accessing the items
```suggestion
            assert data.schedules
            assert data.schedules[i - 1].activated is False
```

Reply:
The test would fail in both cases, and for the given fixture, i expect the items to be present. But you’re right, it’s nicer …